modules/datadict/php/datadict.class.inc

```diff
@@ -58,8 +58,21 @@ class Datadict extends \DataFrameworkMenu
      */
     public function getFieldOptions() : array
     {
+        $instruments     = \Utility::getAllInstruments();
+        $dictInstruments = array();
+        foreach ($instruments as $instrument => $name) {
```

Comment:
```suggestion
        $instruments     = \Utility::getAllInstruments();
        $DB  =& \Database::singleton();
        $activeInstruments = array_filter(
            $allInstruments,
            [$DB, 'tableExists'],
            ARRAY_FILTER_USE_KEY
        );
        $dictInstruments = array();
        foreach ($activeInstruments as $instrument => $name) {
```

Reply:
@jesscall that would automatically eliminate all JSONdata instruments which do not have tables

Reply:
and that would eliminate all instruments which have testnames different then the table names which is exactly to opposite of what I'm trying to do here

Reply:
Very true